geth/jail/jail.go

```diff
@@ -43,6 +48,85 @@ type Jail struct {
 	baseJSCode     string                     // JavaScript used to initialize all new cells with
 }
 
+// Copy returns a new JailCell instance with a new eventloop runtime associated with
+// the given cell.
+func (cell *JailCell) Copy() (common.JailCell, error) {
+	vmCopy := cell.vm.Copy()
+	loCopy := loop.New(vmCopy)
+
+	// Register fetch provider from ottoext.
+	if err := fetch.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	// Register event loop for timers.
+	if err := timers.Define(vmCopy, loCopy); err != nil {
+		return nil, err
+	}
+
+	return &JailCell{
+		id:  cell.id,
+		vm:  vmCopy,
+		lo:  loCopy,
+		sem: semaphore.New(1, JailCellRequestTimeout*time.Second),
+	}, nil
+}
+
+// Fetch attempts to call the underline Fetch API added through the
+// ottoext package.
+func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+	if err := cell.vm.Set("__captureFetch", func(res otto.Value) {
```

Comment:
Ya we can do that for capture but initially i want to json the response but it was not working ad expected.